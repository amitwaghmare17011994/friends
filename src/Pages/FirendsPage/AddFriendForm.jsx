import React, { useState } from 'react'

function AddFriendForm(props) {
    const [error, setError] = useState({})
    const [formState, setFormState] = useState({})
    const { onAdd } = props;
    const onAddFriend = () => {
        if (!formState?.fullName) {
            setError({ ...error, fullName: 'Enter valid full name' })
            return
        }
        if (onAdd) {
            onAdd({...formState})
        }
    }
    const upateFormState = (obj) => setFormState({ ...formState, ...obj })

    const onChange = (e) => {
        const value = e.target.value
        upateFormState({ fullName: value })
        setError({ fullName: value ? '' : error.fullName })
    }

    return (
        <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'baseline' }}>
            <div style={{ marginRight: 20, display: 'flex', flexDirection: 'column', alignItems: 'baseline' }}>
                <input onChange={onChange} placeholder={'Enter full name of friend'} id={'fullname'} />
                {!!error?.fullName &&
                    <span id={'error'} style={{ color: 'red', fontSize: 12 }}>{error?.fullName}</span>}
            </div>
            <button onClick={onAddFriend} id='add_friend'>Add Firend</button>
        </div>
    )
}

export default AddFriendForm
