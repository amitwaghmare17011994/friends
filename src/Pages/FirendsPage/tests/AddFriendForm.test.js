import { create, act } from 'react-test-renderer';
import AddFriendForm from '../AddFriendForm';

describe('Add Friend Form', () => {
    function getRoot(props = {}) {
        return create(<AddFriendForm {...props} />).root

    }
    it('should render form component', () => {
        const root = create(<AddFriendForm />)
        expect(root.toJSON()).toMatchSnapshot();

    })

    it('should have name field', () => {
        const root = getRoot()
        expect(root.findByType('input')).toBeTruthy()

    })

    it('should have add button', () => {
        const root = getRoot()
        expect(root.findByType('button')).toBeTruthy()
    })
    it('should show error enter valid name after click on add', () => {
        const root = getRoot()
        const addButton = root.findByType('button')
        act(() => addButton.props.onClick())
        const errorElement = root.findByProps({ id: 'error' })
        expect(errorElement).toBeTruthy()
    })
    it('on change of full name error should gone', () => {
        const root = getRoot()
        const inputBox = root.findByType('input')
        const addButton = root.findByType('button')
        act(() => addButton.props.onClick())
        act(() => {
            inputBox.props.onChange({ target: { value: 'john' } })
        })
        expect(root.findAllByProps({ id: 'error' }).length).toBe(0)
    })
})