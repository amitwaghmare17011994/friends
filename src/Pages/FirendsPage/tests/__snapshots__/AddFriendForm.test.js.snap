// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Friend Form should render form component 1`] = `
<div
  style={
    Object {
      "alignItems": "baseline",
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "baseline",
        "display": "flex",
        "flexDirection": "column",
        "marginRight": 20,
      }
    }
  >
    <input
      id="fullname"
      onChange={[Function]}
      placeholder="Enter full name of friend"
    />
  </div>
  <button
    id="add_friend"
    onClick={[Function]}
  >
    Add Firend
  </button>
</div>
`;
